ui/apps/platform/src/Containers/Clusters/ClusterPage.tsx

```diff
@@ -181,12 +199,39 @@ function ClusterPage({ clusterId }: ClusterPageProps): ReactElement {
         setPollingCount(pollingCount + 1);
     }, pollingDelay);
 
+    function onChange(path: string, value: boolean | number | string) {
+        // path can be a dot path to property like: tolerationsConfig.disabled
+        setSelectedCluster((oldClusterSettings) => {
+            if (get(oldClusterSettings, path) === undefined) {
+                return oldClusterSettings;
+            }
+
+            const newClusterSettings = cloneDeep(oldClusterSettings);
+            set(newClusterSettings, path, value);
```

Comment:
I noticed the "onChangeLegacy" did things a little differently. Here, it looks like we're passing a new object on every field change. Do you think we should try a similar approach to the legacy one, or did you find the same approach didn't work here?